RecoLocalTracker/Phase2TrackerRecHits/test/RecHitsValidation.cc

```diff
@@ -164,24 +243,112 @@ void Phase2TrackerRecHitsValidation::analyze(const edm::Event& event, const edm:
+	    if (layer<100)
+              histogramLayer->second.globalPosXY[0]->Fill(globalPosClu.z(), globalPosClu.perp());
+	    else
+              histogramLayer->second.globalPosXY[0]->Fill(globalPosClu.x(), globalPosClu.y());
 
             // Pixel module
             if (topol.ncolumns() == 32) {
                 histogramLayer->second.localPosXY[1]->Fill(localPosClu.x(), localPosClu.y());
-                histogramLayer->second.globalPosXY[1]->Fill(globalPosClu.z(), globalPosClu.perp());
+		if (layer<100)
+                  histogramLayer->second.globalPosXY[1]->Fill(globalPosClu.z(), globalPosClu.perp());
+		else
+                  histogramLayer->second.globalPosXY[1]->Fill(globalPosClu.x(), globalPosClu.y());
                 ++nRecHitsPixel;
             }
             // Strip module
             else if (topol.ncolumns() == 2) {
                 histogramLayer->second.localPosXY[2]->Fill(localPosClu.x(), localPosClu.y());
-                histogramLayer->second.globalPosXY[2]->Fill(globalPosClu.z(), globalPosClu.perp());
+		if (layer<100)
+                  histogramLayer->second.globalPosXY[2]->Fill(globalPosClu.z(), globalPosClu.perp());
+		else
+                  histogramLayer->second.globalPosXY[2]->Fill(globalPosClu.x(), globalPosClu.y());
                 ++nRecHitsStrip;
             }
+
+            /*
+	     * Get the cluster from the rechit
+	     */
+	    
+	    const Phase2TrackerCluster1D * clustIt = &*rechitIt->cluster();
+
+            /*
+             * Digis related variables
+             */
+
+            std::vector< unsigned int > clusterSimTrackIds;
+
+            // Get all the simTracks that form the cluster
+            for (unsigned int i(0); i < clustIt->size(); ++i) {
+                unsigned int channel(Phase2TrackerDigi::pixelToChannel(clustIt->firstRow() + i, clustIt->column())); 
+                unsigned int simTrackId(getSimTrackId(pixelSimLinks, detId, channel));
+                clusterSimTrackIds.push_back(simTrackId);
+            }
+
+            /*
+             * SimHits related variables
+             */
+
+            unsigned int primarySimHits(0);
+            unsigned int otherSimHits(0);
+
+            for (unsigned int simhitidx = 0; simhitidx < 2; ++simhitidx) {
+              for (edm::PSimHitContainer::const_iterator hitIt(simHitsRaw[simhitidx]->begin()); hitIt != simHitsRaw[simhitidx]->end(); ++hitIt) {
+                if (rawid == hitIt->detUnitId() and std::find(clusterSimTrackIds.begin(), clusterSimTrackIds.end(), hitIt->trackId()) != clusterSimTrackIds.end()) {
+                    Local3DPoint localPosHit(hitIt->localPosition());
+
+                    histogramLayer->second.deltaXRecHitsSimHits[0]->Fill(localPosClu.x() - localPosHit.x());
+                    histogramLayer->second.deltaYRecHitsSimHits[0]->Fill(localPosClu.y() - localPosHit.y());
+                    // Pixel module
+                    if (topol.ncolumns() == 32) {
```

Comment:
Hi @davidlange6 , this code was copied out of RecoLocalTracker/SiPhase2Clusterizer/test/ClustersValidation.cc, and I suspect before that it came out of the digitizer code (as @venturia alluded to).
This is just validation code (and to be replaced midterm with DQM validation). I'm not sure if you're asking me to do anything. It seems your suggestion should go on a DPG todo list.